### PR TITLE
DApp/core update

### PR DESCRIPTION
**Pull request by nickreynolds**
- Reorganizing civilTCR to use separate classes to get data for Listing, Challenge, etc
- expose helper functions to determine state of Listing, rather than using ListingState function
- Adding functionality to DApp including ability to apply to TCR
- Updating TransactionButton to be more descriptive of 2-step process
- Adding CountdownTimer component

### CODE REVIEW

**Comment by nickreynolds** [comment]
next PR will include moving all ListingData calls/state into Redux/actions

File: packages/dapp/src/apis/civilTCR.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  return tcr.requestAppeal(address);
+}
+
+export async function resolveChallenge(address: EthAddress): Promise<TwoStepEthTransaction> {
+  const civil = getCivil();
+  const tcr = civil.tcrSingletonTrusted();
+  return tcr.updateListing(address);
+}
+
+export async function getNewsroom(address: EthAddress): Promise<any> {
```
Comment (ritave) [inline packages/dapp/src/apis/civilTCR.ts]: Might as well export all the wrappers in core

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "test": "lerna run test --stream --sort",
     "lerna:run": "lerna run",
     "prepare": "lerna run prepare --no-sort",
-    "ganache": "run-p ganache:raw 'lerna:run --scope @joincivil/contracts migrate:ganache'",
+    "ganache": "run-p ganache:raw",
```
[the remark this answers is not in the document]
Comment (walfly) [reply package.json]: i think we can't add `--reset` unless we put it in the name of the command

File: packages/core/src/contracts/tcr/listing.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
+import { Observable, Subscription } from "rxjs";
+import "@joincivil/utils";
+import { DecodedLogEntryEvent } from "@joincivil/typescript-types";
+
```
Comment (walfly) [inline packages/core/src/contracts/tcr/listing.ts]: whitespace

File: packages/core/src/utils/listingDataHelpers/appealChallengeHelper.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { isInCommitStage, isInRevealStage, isVotePassed } from "./pollHelper";
+import { AppealChallengeData } from "../../types";
+
+/**
+ * Checks in an Appeal Challenge is in the Commit stage
+ * @param challengeData the AppealChallengeData to check
+ */
+export function isAppealChallengeInCommitStage(challengeData: AppealChallengeData): boolean {
```
[the remark this answers is not in the document]
Comment (walfly) [reply packages/core/src/utils/listingDataHelpers/appealChallengeHelper.ts]: these don't pollute global scope, they're module variables.  I think our code could be improved by relying more on modules and less on classes being all super object-oriented is fighting javascript a bit.

**Comment by nickreynolds** [comment]
I merged this without making the static factory / helper function changes so that I could unblock people for tomorrow morning. I did the entire refactor, but chose to revert it based on Walker's comments and how we think core will best interact with the DApp. If we eventually decide to make those changes, the refactor won't take very long.

**Pull request merged**